src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -261,6 +284,45 @@ class RegularSync(
     }
   }
 
+  private def requestMissingNode(nodeId: ByteString): Unit = {
+    bestPeer match {
+      case Some(peer) =>
+        waitingForActor = Some(context.actorOf(
+          PeerRequestHandler.props[GetNodeData, NodeData](
+            peer, peerResponseTimeout, etcPeerManager, peerEventBus,
+            requestMsg = GetNodeData(List(nodeId)),
+            responseMsgCode = NodeData.code)))
+
+      case None =>
+        log.debug("Requesting missing state nodes: no peers to download from")
+        scheduleResume()
+    }
+  }
+
+  private def handleRedownloadedStateNodes(peer: Peer, nodes: Seq[ByteString]): Unit = {
+    if (nodes.isEmpty) {
+      log.debug(s"Did not receive missing state node from peer ($peer)")
+      resumeWithDifferentPeer(peer, "did not receive missing state node")
+    } else {
+      val MissingStateNodeRetry(requestedHash, blockPeer, blocksToRetry) = missingStateNodeRetry.get
```

Comment:
`.get` alert :smile: someone may just send us NodeData msg and cause this actor to restart

Reply:
Nb. `RegularSync` actor does not restart

Reply:
what happens when uncaught exception is thrown? I thought default supervisorStrategy is to restart

Reply:
My observation was that after `MPTException` is thrown, the `RegularSync` actor does not restart

Reply:
strange...wasn't it wrapped in Try or something like that in Ledger?

Reply:
For me it seems that after the exception, `RegularSync` actor stops asking for blocks, does that make sense or am I seeing something wrong?

Reply:
There is no logging whatsoever in `RegularSync` after an exception is thrown, so I just assumed the actor died and was not restarted. Whatever the true nature of this behaviour is, it **was not changed** in this PR. I also don't see any reason to change it.

Reply:
mmm...but this was changed. After this PR this would break if we didn't have an MPT node (our fault) but in this case, any malicious peer can start sending us MPT node messages to make our `RegularSync` to stop, can't they?

Reply:
> mmm...but this was changed.

The way `RegularSync` actor (or its supervisor) deals with exceptions was not changed.

> peer can start sending us MPT node messages to make our RegularSync to stop, can't they?

They can't. That was fixed as per [this comment](https://github.com/input-output-hk/mantis/pull/365#discussion_r156730881)

Reply:
you are going to kill me...I didn't pull that change.